manager/assets/modext/util/utilities.js

```diff
@@ -82,6 +82,9 @@ Ext.override(Ext.form.NumberField, {
 Ext.override(Ext.form.BasicForm,{
     clearDirty : function(nodeToRecurse){
         nodeToRecurse = nodeToRecurse || this;
+        if (!nodeToRecurse || !nodeToRecurse.items || typeof nodeToRecurse.items.each !== 'function') {
+            return;
+        }
         nodeToRecurse.items.each(function(f){
```

Comment:
I believe you could use optional chaining to simplify this to:
```suggestion
        nodeToRecurse?.items?.each(function(f){
```

Reply:
Also thought of that, but as in the docs it states that the [manager also supports Internet Explorer 11](https://docs.modx.com/3.x/en/getting-started/server-requirements), I don't think we can use ultra-modern JavaScript. The [Internet Explorer does not support](https://caniuse.com/?search=optional%20chaining) optional chaining.

Reply:
We're already using optional chaining as well as other ES6+ features elsewhere in the code. The support page probably needs updating, especially since IE was officially retired by Microsoft as of June last year.

Reply:
Well, in that case 😃